EIPS/EIP-X-metadata-nickjohnson.md

```diff
@@ -0,0 +1,72 @@
+## Preamble
+
+    EIP: <to be assigned>
+    Title: Address metadata registry
+    Author: Nick Johnson <nick@ethereum.org>
+    Type: Standard track
+    Category: ERC
+    Status: Draft
+    Created: 2018-03-12
+
+## Abstract
+This EIP specifies a registry for address metadata, permitting both contracts and external accounts to supply metadata about themselves to onchain and offchain callers. This permits use-cases such as generalised authorisations, providing token acceptance settings, and claims registries.
+
+## Motivation
+An increasing set of use cases require storage of metadata associated with an address; see for instance EIP 777 and EIP 780, and the ENS reverse registry in EIP 181. Presently each use-case defines its own specialised registry. To prevent a proliferation of special-purpose registry contracts, we instead propose a single standardised registry using an extendable architecture that allows future standards to implement their own metadata standards.
+
+## Specification
+The metadata registry has the following interface:
+```
+interface AddressMetadataRegistry {
+  function provider(address target) view returns(address);
+  function setProvider(address _provider);
+}
+```
+
+`setProvider` specifies the metadata registry to be associated with the caller's address, while `provider` returns the address of the metadata registry for the supplied address.
+
+The metadata registry will be compiled with an agreed-upon version of Solidity and deployed using the trustless deployment mechanism to a fixed address that can be replicated across all chains.
+
+## Provider specification
+
+Providers may implement any subset of the metadata record types specified here. Where a record types specification requires a provider to provide multiple functions, the provider MUST implement either all or none of them. Providers MUST throw if called with an unsupported function ID.
+
+Providers have one mandatory function:
+
+```
+function supportsInterface(bytes4 interfaceID) constant returns (bool)
+```
+
+The `supportsInterface` function is documented in [EIP 165](https://github.com/ethereum/EIPs/issues/165), and returns true if the provider implements the interface specified by the provided 4 byte identifier. An interface identifier consists of the XOR of the function signature hashes of the functions provided by that interface; in the degenerate case of single-function interfaces, it is simply equal to the signature hash of that function. If a provider returns `true` for `supportsInterface()`, it must implement the functions specified in that interface.
```

Comment:
After reading EIP 165 (draft rather than the GitHub issue, which was much harder to follow), I am now confused why this entire section exists.  It appears to just be re-hashing what EIP 165 says, but with less context, less detail and fewer examples.  Why not just say that providers are defined by EIP 165 and leave it at that?

Reply:
I thought a brief description would be useful so that someone who only wants to implement it doesn't need to read all of 165, too.

Reply:
I need to step away and think a bit more to avoid curve fitting my personal experience, but my gut right now is telling me that the fact that there was a bunch of text rather than just a link resulted in me ignoring the link and reading just the text, which unfortunately is lacking in the detail I would expect from an EIP.